x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/regex/LikePattern.java

```diff
@@ -48,6 +48,16 @@ public String asJavaRegex() {
         return regex;
     }
 
+    @Override
+    public boolean matchesAll() {
+        for (int i = 0 ; i < pattern.length(); i++) {
+            if (pattern.charAt(i) != '*') {
```

Comment:
Isn't the wildcard character a `%`? To my understanding the replacement of `%` -> `*` happens later here: https://github.com/elastic/elasticsearch/pull/62682/files#diff-cb6f05d0222f3291ea3df5d33525d6f6R80

Reply:
You are correct. Fixed this.